### ISSUE TITLE
A few question regarding your implementation

### ISSUE THREAD

**Issue by OLibykov** [WenjieZhao1/S2M#14]
Hi,
I have a few questions/comments/clarifications regarding your implementation. I read the Detectron2 documentation and compared it with what's written in your paper and in S2M/configs/OE/OE.yaml. In the "Implementation Details" section of the paper, you write:

During the training process, we initiate the learning rate at 1 × 10⁻⁴, employing a learning rate adjustment strategy. This approach incrementally raises the learning rate in a linear fashion to reach 2.5 × 10⁻³ over the course of 1000 iterations. We train our prompt detector on 1 × NVIDIA RTX A5000 GPU within about 10 hours for 100 epochs.

1. Why does OE.yaml contain BASE_LR: 0.00025 = 2.5e-4, but you override it in the command:
python3 plain_train_net.py --config-file ../configs/OE/OE.yaml --num-gpus 1 SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.0025
This makes BASE_LR = 2.5e-3. So why is there incorrect information in OE.yaml?

2. You write that the initial learning rate is 1e-4, but this is not specified in OE.yaml. By default, in S2M/detectron2/config/defaults.py, the following parameters are set:
_C.SOLVER.WARMUP_FACTOR = 1.0 / 1000
_C.SOLVER.WARMUP_ITERS = 1000
_C.SOLVER.WARMUP_METHOD = "linear"
As I understand it, the warmup strategy means the model starts training with
lr = SOLVER.BASE_LR * WARMUP_FACTOR = 2.5e-6
and linearly increases to 2.5e-3 over 1000 iterations. So where does the 1e-4 in the paper come from?

3. The paper says you trained for 100 epochs. Detectron2 doesn’t specify the number of epochs directly, but we can compute it:
SOLVER.MAX_ITER = 148300 (from OE.yaml)  
SOLVER.IMS_PER_BATCH = 4 (from OE.yaml)  
n_train_object = 2966  
n_epoch = SOLVER.MAX_ITER * SOLVER.IMS_PER_BATCH / n_train_object = 200 epochs  
So how many epochs did you actually train for?

4. You set IMS_PER_BATCH = 4 (default is 16). Did you reduce it to make it fit into the memory of the NVIDIA RTX A5000 GPU?

I would really appreciate your answers.

**Comment by WenjieZhao1** [comment]
The parameters used here may not be exactly the same as those in the paper.
